src/common/signalk/SKNMEA2000Parser.cpp

```diff
@@ -43,19 +46,170 @@ const SKUpdate& SKNMEA2000Parser::parse(const SKSourceInput& input, const tN2kMs
+  unsigned char sid;          // Sequence ID
+  uint16_t SystemDate;        // Days since 1970-01-01
+  double SystemTime;          // seconds since midnight mit 2 Kommastellen
+  tN2kTimeSource TimeSource;  // see tN2kTimeSource (z.B.: GPS)
+
+    if (ParseN2kSystemTime(msg,sid,SystemDate,SystemTime,TimeSource) ) {
+
+      /*
+      // static global variables needed....
+      if ( SystemDate > 0 && SystemTime > 0 && dateTime::timeStampNeeded ) {
+        // time function class
+        dt.setTimeStamp( &SystemDate, &SystemTime);
+        // the first 20 Seconds try to get date and time from N2k bus
+        if ( millis() > 20000 ) dateTime::timeStampNeeded = false;
+      }
+      */
+    } else {
+      INFO("Unable to parse NMEA2000 with PGN %i", msg.PGN);
+    }
+  // nothing todo at the moment
+  return _invalidSku;
+}
+
+//  ***********************************************
+//   PGN 127245 Rudder
+//        →  sid
+//        →  RudderPosition [rad]
+// *  ********************************************** */
+const SKUpdate& SKNMEA2000Parser::parse127245(const SKSourceInput& input, const tN2kMsg& msg, const SKTime& timestamp) {
+  unsigned char Instance;
+  tN2kRudderDirectionOrder RudderDirectionOrder;
+  double RudderPosition;
+  double AngleOrder;
+
+  if (ParseN2kRudder(msg,RudderPosition,Instance,RudderDirectionOrder,AngleOrder) ) {
+    SKUpdateStatic<2> *update = new SKUpdateStatic<2>();
+    update->setTimestamp(timestamp);
+
+    SKSource source = SKSource::sourceForNMEA2000(input, msg.PGN, msg.Priority, msg.Source);
+    update->setSource(source);
+    update->setSteeringRudderAngle(RudderPosition);
+    DEBUG("Rudder: %d",SKRadToDeg(RudderPosition));
+    _sku = update;
+    return *_sku;
   }
-  // Water Depth
-  if (msg.PGN == 128267) {
-    return parse128267(input, msg, timestamp);
+  else {
+    INFO("Unable to parse NMEA2000 with PGN %i", msg.PGN);
   }
+  // nothing todo at the moment....
+  return _invalidSku;
+}
+
+// *****************************************************************************
+//   PGN 127250 VESSEL HEADING RAPID
+// *****************************************************************************
+const SKUpdate& SKNMEA2000Parser::parse127250(const SKSourceInput& input, const tN2kMsg& msg, const SKTime& timestamp) {
+  unsigned char sid;
+  tN2kHeadingReference HeadingReference;
+  double Heading = 361; // not valid value
+  double Deviation = 0; // NAN would be better!
+  double Variation = 0; // NAN would be better!
 
-  DEBUG("No known conversion for PGN %i", msg.PGN);
+  if (ParseN2kHeading(msg,sid,Heading,Deviation,Variation,HeadingReference) ) {
+
+    if ( String(HeadingReference) == "magnetic" ) {
+      // Deviation and Variation could be sent
+      // (Timo's NMEA Simulator is sending Dev and Var at magnetic HDG only)
+      Heading += Deviation + Variation;
+    }
+    // if ( String(HeadingReference) == "true" ) {  }
+
+    SKUpdateStatic<2> *update = new SKUpdateStatic<2>();
```

Comment:
This causes a memory leak because you are creating a new `SKUpdateStatic` but you are not keeping the pointer. You should assign it to `_sku` immediately to make sure we do not "lose" it.

If you assign it to `_sku` the `parse()` method will delete it next time a NMEA2000 message is received.